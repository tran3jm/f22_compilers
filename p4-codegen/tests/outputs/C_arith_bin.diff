--- outputs/C_arith_bin.txt	2022-11-18 00:32:30.439406870 -0500
+++ expected/C_arith_bin.txt	2022-11-17 23:32:01.737740687 -0500
@@ -2,24 +2,24 @@
   push BP
   i2i SP => BP
   addI SP, -32 => SP
-  loadI 3 => r1
-  loadI 4 => r2
-  add r1, r2 => r3
-  storeAI r3 => [BP-8]
-  loadI 3 => r5
-  loadI 4 => r6
-  sub r5, r6 => r7
-  storeAI r7 => [BP-16]
+  loadI 3 => r0
+  loadI 4 => r1
+  add r0, r1 => r2
+  storeAI r2 => [BP-8]
+  loadI 3 => r3
+  loadI 4 => r4
+  sub r3, r4 => r5
+  storeAI r5 => [BP-16]
+  loadI 39 => r6
+  loadI 2 => r7
+  mult r6, r7 => r8
+  storeAI r8 => [BP-24]
   loadI 39 => r9
   loadI 2 => r10
-  mult r9, r10 => r11
-  storeAI r11 => [BP-24]
-  loadI 39 => r13
-  loadI 2 => r14
-  div r13, r14 => r15
-  storeAI r15 => [BP-32]
-  loadAI [BP-8] => r16
-  i2i r16 => RET
+  div r9, r10 => r11
+  storeAI r11 => [BP-32]
+  loadAI [BP-8] => r12
+  i2i r12 => RET
   jump l0
 l0:
   i2i BP => SP
