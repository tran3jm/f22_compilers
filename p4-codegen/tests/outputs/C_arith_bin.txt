main:
  push BP
  i2i SP => BP
  addI SP, -32 => SP
  loadI 3 => r1
  loadI 4 => r2
  add r1, r2 => r3
  storeAI r3 => [BP-8]
  loadI 3 => r5
  loadI 4 => r6
  sub r5, r6 => r7
  storeAI r7 => [BP-16]
  loadI 39 => r9
  loadI 2 => r10
  mult r9, r10 => r11
  storeAI r11 => [BP-24]
  loadI 39 => r13
  loadI 2 => r14
  div r13, r14 => r15
  storeAI r15 => [BP-32]
  loadAI [BP-8] => r16
  i2i r16 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 7
