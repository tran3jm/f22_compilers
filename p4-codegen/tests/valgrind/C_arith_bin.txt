==2079904== Memcheck, a memory error detector
==2079904== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2079904== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==2079904== Command: ../decaf inputs/arith_bin.decaf
==2079904== 
main:
  push BP
  i2i SP => BP
  addI SP, -32 => SP
  loadI 3 => r1
  loadI 4 => r2
  add r1, r2 => r3
  storeAI r3 => [BP-8]
  loadI 3 => r5
  loadI 4 => r6
  sub r5, r6 => r7
  storeAI r7 => [BP-16]
  loadI 39 => r9
  loadI 2 => r10
  mult r9, r10 => r11
  storeAI r11 => [BP-24]
  loadI 39 => r13
  loadI 2 => r14
  div r13, r14 => r15
  storeAI r15 => [BP-32]
  loadAI [BP-8] => r16
  i2i r16 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 7
==2079904== 
==2079904== HEAP SUMMARY:
==2079904==     in use at exit: 0 bytes in 0 blocks
==2079904==   total heap usage: 2,205 allocs, 2,205 frees, 509,920 bytes allocated
==2079904== 
==2079904== All heap blocks were freed -- no leaks are possible
==2079904== 
==2079904== For lists of detected and suppressed errors, rerun with: -s
==2079904== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
