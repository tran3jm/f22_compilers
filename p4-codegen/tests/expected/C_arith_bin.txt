main:
  push BP
  i2i SP => BP
  addI SP, -32 => SP
  loadI 3 => r0
  loadI 4 => r1
  add r0, r1 => r2
  storeAI r2 => [BP-8]
  loadI 3 => r3
  loadI 4 => r4
  sub r3, r4 => r5
  storeAI r5 => [BP-16]
  loadI 39 => r6
  loadI 2 => r7
  mult r6, r7 => r8
  storeAI r8 => [BP-24]
  loadI 39 => r9
  loadI 2 => r10
  div r9, r10 => r11
  storeAI r11 => [BP-32]
  loadAI [BP-8] => r12
  i2i r12 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 7
